docs/cookbook/entities/custom-model-accessible-for-channel-admin.rst

```diff
@@ -0,0 +1,289 @@
+.. rst-class:: plus-doc
+
+How to add a custom model accessible for respective channel administrators?
+===========================================================================
+
+Given that you are using Sylius Plus, the licensed edition of Sylius, you may have
+the Administrators per Channel defined in your application. Thus when you add a new,
+channel-based entity to it, you will need to enable this entity to be viewed only by the relevant channel admins.
+
+1. Define your custom model, our example will be the **Supplier entity**
+------------------------------------------------------------------------
+
+In order to prepare a simple Entity follow :doc:`this guide </cookbook/entities/custom-model>`.
+
+* Having your Supplier entity created, add a channel field with relation to the ``Channel`` entity:
+
+.. code-block:: php
+
+    /**
+      * @var ChannelInterface
+      * @ORM\ManyToOne(targetEntity="Sylius\Plus\Entity\ChannelInterface")
+      * @ORM\JoinColumn(name="channel_id", referencedColumnName="id", nullable=true)
+      */
+     protected $channel;
+
+* Create a proper migration for these changes.
+
+* Next, create a form type for your entity:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Form;
+
+    use Sylius\Bundle\ChannelBundle\Form\Type\ChannelChoiceType;
+    use Sylius\Bundle\ResourceBundle\Form\Type\AbstractResourceType;
+    use Sylius\Plus\ChannelAdmin\Application\Provider\AvailableChannelsForAdminProviderInterface;
+    use Symfony\Component\Form\Extension\Core\Type\TextType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    final class SupplierType extends AbstractResourceType
+    {
+        /** @var AvailableChannelsForAdminProviderInterface */
+        private $availableChannelsForAdminProvider;
+
+        public function __construct(
+            AvailableChannelsForAdminProviderInterface $availableChannelsForAdminProvider,
+            string $dataClass,
+            array $validationGroups = []
+        ) {
+            parent::__construct($dataClass, $validationGroups);
+
+            $this->availableChannelsForAdminProvider = $availableChannelsForAdminProvider;
+        }
+
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('name', TextType::class, [
+                    'label' => 'Name'
+                ])
+                ->add('channel', ChannelChoiceType::class, [
+                    'choices' => $this->availableChannelsForAdminProvider->getChannels(),
+                    'label' => 'sylius.ui.channel',
+                ])
+            ;
+        }
+
+        public function getBlockPrefix(): string
+        {
+            return 'supplier';
+        }
+    }
+
+.. code-block:: yaml
+
+    App\Form\SupplierType:
+        arguments: ['App\Entity\Supplier', ['sylius'], '@Sylius\Plus\ChannelAdmin\Application\Provider\AvailableChannelsForAdminProviderInterface']
```

Comment:
```suggestion
        arguments: ['@Sylius\Plus\ChannelAdmin\Application\Provider\AvailableChannelsForAdminProviderInterface', 'App\Entity\Supplier', ['sylius']]
```
Shouldn't the arguments here be in the same order as in the class's constructor?

Reply:
IMO, the arguments in constructor should be in different order